File: Data Structures & Algorithms/concatenation-of-array/submission-0.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * (2*n)
        index = 0
        for i in nums:
            ans[index]=i
            ans[index+n]=i
            index+=1
        return ans

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {

        let ans = new Array(nums.length);
        let product = 1;
        let zeroCnt = 0;

        for (let i=0; i<nums.length; i++){
            if(nums[i]!==0){
                product *= nums[i];

            }else{
                zeroCnt++;
            }
        }

        if(zeroCnt>1){
            return Array(nums.length).fill(0);
        }

        for (let i=0; i<nums.length; i++){
            if(zeroCnt>0){
                ans[i] = nums[i] === 0? product : 0;
            }else{
                ans[i] = product/ nums[i]
            }
        }
        return ans;

    }
}
